cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaMirrorMaker2Connectors.java

```diff
@@ -324,4 +327,23 @@ private static String addTLSConfigToMirrorMaker2ConnectorConfig(Map<String, Obje
             return "PLAINTEXT";
         }
     }
+
+    private static boolean hasMatchingBootstrapServers(KafkaMirrorMaker2 kafkaMirrorMaker2, KafkaMirrorMaker2MirrorSpec mirror, String connectCluster) {
+        Map<String, String> configs = new HashMap<>();
+
+        Optional.ofNullable(kafkaMirrorMaker2)
+                .map(KafkaMirrorMaker2::getSpec)
+                .map(KafkaMirrorMaker2Spec::getClusters)
+                .orElse(Collections.emptyList())
+                .forEach(clusterLists -> {
+                    if (connectCluster != null && connectCluster.equals(clusterLists.getAlias())) {
+                        configs.put("connectClusterBootstrapServer", clusterLists.getBootstrapServers());
+                    }
+                    if (mirror.getTargetCluster() != null && mirror.getTargetCluster().equals(clusterLists.getAlias())) {
+                        configs.put("targetClusterBootstrapServers", clusterLists.getBootstrapServers());
+                    }
+                });
+
+        return configs.get("connectClusterBootstrapServer").equals(configs.get("targetClusterBootstrapServers"));
```

Comment:
This seems a very weird way to do it. You know the connect cluster alias (bootstrap server) and you can simply compare it to the target clusters in for loop you do not need to create a new Hashmap for it or anything like that.

The code also assumes only one target cluster, but there can be more, or?

Reply:
The cluster alias is different from the bootstrap server, so we have a List of clusters and two aliases. So the code has to find the cluster object for each alias, get the bootstrap server out for each and then compare those two bootstrap servers. There are multiple clusters, but each mirror only has one target cluster.

Reply:
A nicer way to do it might be to use the `distinct` method:

```
Optional.ofNullable(kafkaMirrorMaker2)
    .map(KafkaMirrorMaker2::getSpec)
    .map(KafkaMirrorMaker2Spec::getClusters)
    .orElse(Collections.emptyList())
    .stream()
    .filter(cluster -> connectClusterAlias.equals(cluster.getAlias()) || targetAlias.equals(cluster.getAlias()))
    .map(KafkaMirrorMaker2ClusterSpec::getBootstrapServers)
    .distinct()
    .toList();
```

Then look at the size of the list, or even just use `count()` directly